jsonschema_lexer/_lexer.py

```diff
@@ -24,109 +25,181 @@ class JSONSchemaLexer(JsonLexer):
     ]
 
     data_types: ClassVar[list[str]] = [
-        "object",
-        "integer",
-        "string",
-        "number",
-        "array",
-        "boolean",
-        "null",
-    ]
-    core_keywords: ClassVar[list[str]] = [
-        "$schema",
-        "$id",
-        "$ref",
-        "$defs",
-        "$comment",
-        "$dynamicAnchor",
-        "$dynamicRef",
-        "$anchor",
-        "$vocabulary",
-    ]
-    applicator_keywords: ClassVar[list[str]] = [
-        "oneOf",
-        "allOf",
-        "anyOf",
-        "if",
-        "then",
-        "else",
-        "not",
-        "properties",
-        "patternProperties",
-        "additionalProperties",
-        "dependentSchemas",
-        "propertyNames",
-        "prefixItems",
-        "contains",
-        "items",
-    ]
-    meta_data_keywords: ClassVar[list[str]] = [
-        "title",
-        "description",
-        "default",
-        "deprecated",
-        "examples",
-        "readOnly",
-        "writeOnly",
-    ]
-    validation_keywords: ClassVar[list[str]] = [
-        "type",
-        "enum",
-        "const",
-        "minLength",
-        "maxLength",
-        "pattern",
-        "maximum",
-        "exclusiveMinimum",
-        "multipleOf",
-        "exclusiveMaximum",
-        "minimum",
-        "dependentRequired",
-        "minProperties",
-        "maxProperties",
-        "required",
-        "minItems",
-        "maxItems",
-        "minContains",
-        "maxContains",
-        "uniqueItems",
-    ]
-    other_keywords: ClassVar[list[str]] = [
-        "format",
-        "unevaluatedItems",
-        "unevaluatedProperties",
-        "contentEncoding",
-        "contentMediaType",
-        "contentSchema",
-        "format_assertion",
+        '"object"',
+        '"integer"',
+        '"string"',
+        '"number"',
+        '"array"',
+        '"boolean"',
+        '"null"',
     ]
 
-    parsed_keywords: ClassVar[list[str]] = [
-        '"%s"' % keyword
-        for keyword in (
-            core_keywords
-            + applicator_keywords
-            + meta_data_keywords
-            + validation_keywords
-            + other_keywords
+    def get_dialect_keywords(self, dialect_url: str | None) -> list[str]:
+        match dialect_url:
+            case '"https://json-schema.org/draft/2020-12/schema"':
```

Comment:
I'd probably move these to globals even, and just leave selecting between the globals to the lexer.